packages/backend/server/tests/nestjs/throttler.spec.ts

```diff
@@ -0,0 +1,260 @@
+  await t.context.app.close();
+});
+
+function rateLimitHeaders(res: Response) {
+  return {
+    limit: res.header['x-ratelimit-limit'],
+    remaining: res.header['x-ratelimit-remaining'],
+    reset: res.header['x-ratelimit-reset'],
+    retryAfter: res.header['retry-after'],
+  };
+}
+
+test('should be able to prevent requests if limit is reached', async t => {
+  const { app } = t.context;
+
+  const stub = Sinon.stub(app.get(ThrottlerStorage), 'increment').resolves({
+    timeToExpire: 10,
+    totalHits: 21,
+  });
+  const res = await request(app.getHttpServer())
+    .get('/nonthrottled/strict')
+    .expect(HttpStatus.TOO_MANY_REQUESTS);
+
+  const headers = rateLimitHeaders(res);
+
+  t.is(headers.retryAfter, '10');
+
+  stub.restore();
+});
+
+// ====== unauthenticated user visits ======
+test('should use default throttler for unauthenticated user when not specified', async t => {
+  const { app } = t.context;
+
+  const res = await request(app.getHttpServer())
+    .get('/nonthrottled/default')
+    .expect(200);
+
+  const headers = rateLimitHeaders(res);
+
+  t.is(headers.limit, '120');
+  t.is(headers.remaining, '119');
+  t.is(headers.reset, '60');
+});
+
+test('should skip throttler for unauthenticated user when specified', async t => {
+  const { app } = t.context;
+
+  const res = await request(app.getHttpServer())
+    .get('/nonthrottled/skip')
+    .expect(200);
+
+  const headers = rateLimitHeaders(res);
+
+  t.is(headers.limit, undefined!);
+  t.is(headers.remaining, undefined!);
+  t.is(headers.reset, undefined!);
+});
+
+test('should use specified throttler for unauthenticated user', async t => {
+  const { app } = t.context;
+
+  const res = await request(app.getHttpServer())
+    .get('/nonthrottled/strict')
+    .expect(200);
+
+  const headers = rateLimitHeaders(res);
+
+  t.is(headers.limit, '20');
+  t.is(headers.remaining, '19');
+  t.is(headers.reset, '60');
+});
+
+// ==== authenticated user visits ====
+test('should not protect unspecified routes', async t => {
+  const { app, cookie } = t.context;
+
+  const res = await request(app.getHttpServer())
+    .get('/nonthrottled/default')
+    .set('Cookie', cookie)
+    .expect(200);
+
+  const headers = rateLimitHeaders(res);
+
+  t.is(headers.limit, undefined!);
+  t.is(headers.remaining, undefined!);
+  t.is(headers.reset, undefined!);
+});
+
+test('should use default throttler for authenticated user when not specified', async t => {
+  const { app, cookie } = t.context;
+
+  const res = await request(app.getHttpServer())
+    .get('/throttled/default')
+    .set('Cookie', cookie)
+    .expect(200);
+
+  const headers = rateLimitHeaders(res);
+
+  t.is(headers.limit, '120');
+  t.is(headers.remaining, '119');
+  t.is(headers.reset, '60');
+});
+
+test('should skip throttler for authenticated user when specified', async t => {
+  const { app, cookie } = t.context;
+
+  const res = await request(app.getHttpServer())
+    .get('/throttled/skip')
```

Comment:
what about the unauthenticated user with skip decorator in throttled resolver?

Reply:
L150

Reply:
L150 just tested without @Throttle()

Reply:
L164